### PR TITLE
pvc_create_delete refactoring

### PR DESCRIPTION
**Pull request by Avilir**
This PR improve the TC runtime and the memory consumption during the test.
The Original TC ran for 5 Hours and 45 Min with memory consumption of up to 2.2 GiB of memory, while after this PR
the TC will run for only 1 Hour and 15 Min, with memory consumption of up to 380 MiB only.

as can be seen in this graph: Screen Shot 2022-03-27 at 11 35 22

This PR also:
* reduce duplicated code
* improvement of the ES report
* fix issue with the ES report, said it written even if not

Signed-off-by: Avi Layani <alayani@redhat.com>

### CODE REVIEW

File: ocs_ci/helpers/performance_lib.py [hunk 1]
```diff
@@ -375,3 +380,210 @@ def csi_bulk_pvc_time_measure(interface, pvc_objs, operation, start_time):
...
+                                ] = calculate_operation_time(
+                                    name, results[name]["csi_delete"]
+                                )
+
+    logger.debug(f"All results are : {json.dumps(results, indent=3)}")
+    return results
+
+
+def wait_for_resource_bulk_status(
+    resource, resource_count, namespace, status, timeout=60, sleep_time=3
```
Comment (OdedViner) [inline ocs_ci/helpers/performance_lib.py:551]: Can you use `TimeoutSampler`?

Comment (Avilir) [reply ocs_ci/helpers/performance_lib.py:551]: i try to avoid using object to reduce memory consumption as posible.

**Comment by Avilir** [comment]
it is look like test failed on network issue, from the test log :
```
2022-03-27 17:27:54  14:27:54 - MainThread - ocs_ci.utility.utils - WARNING - Command stderr: Unable to connect to the server: dial tcp: lookup api.jnk-pr5636-b2537.qe.rh-ocs.com on 10.5.30.160:53: read udp 10.0.207.179:54694->10.5.30.160:53: i/o timeout
2022-03-27 17:27:54  
2022-03-27 17:27:54  14:27:54 - MainThread - ocs_ci.ocs.ocp - WARNING - Failed to get resource: pvc-test-76bf7293561a498e97c095b7907256b of kind: PersistentVolumeClaim, selector: None, Error: Error during execution of command: oc -n pas-test-namespace get PersistentVolumeClaim pvc-test-76bf7293561a498e97c095b7907256b -n pas-test-namespace -o yaml.
2022-03-27 17:27:54  Error is Unable to connect to the server: dial tcp: lookup api.jnk-pr5636-b2537.qe.rh-ocs.com on 10.5.30.160:53: read udp 10.0.207.179:54694->10.5.30.160:53: i/o timeout
2022-03-27 17:27:54  
2022-03-27 17:27:54  14:27:54 - MainThread - ocs_ci.ocs.ocp - WARNING - Number of attempts to get resource reached!
2022-03-27 17:27:54  14:27:54 - MainThread - ocs_ci.helpers.helpers - INFO - Creating directory /home/jenkins/current-cluster-dir/logs/failed_testcase_ocs_logs_1648388371/test_multiple_pvc_deletion_measurement_performance[CephBlockPool]_performance_stats
```

triggering new Validation Job.

**Comment by Avilir** [comment]
Verified locally.

**Review by ypersky1980 — APPROVED** [review]
LGTM, and the time improvement is amazing!

**Comment by keemano** [comment]
LGTM

**Review by ramkiperiy — APPROVED** [review]

**Pull request merged**